controllers/k8ssandra/cleanup.go

```diff
@@ -350,129 +184,35 @@ func (r *K8ssandraClusterReconciler) findDcForDeletion(
 	return nil, nil, nil
 }
 
-func (r *K8ssandraClusterReconciler) deleteK8ssandraConfigMaps(
-	ctx context.Context,
-	kc *k8ssandraapi.K8ssandraCluster,
-	dcTemplate k8ssandraapi.CassandraDatacenterTemplate,
-	namespace string,
-	remoteClient client.Client,
-	kcLogger logr.Logger,
-) (hasErrors bool) {
-	selector := k8ssandralabels.CleanedUpByLabels(client.ObjectKey{Namespace: kc.Namespace, Name: kc.SanitizedName()})
-	configMaps := &corev1.ConfigMapList{}
-	options := client.ListOptions{
-		Namespace:     namespace,
-		LabelSelector: labels.SelectorFromSet(selector),
-	}
-	if err := remoteClient.List(ctx, configMaps, &options); err != nil {
-		kcLogger.Error(err, "Failed to list ConfigMap objects", "Context", dcTemplate.K8sContext)
-		return true
-	}
-	for _, rp := range configMaps.Items {
-		if err := remoteClient.Delete(ctx, &rp); err != nil {
-			key := client.ObjectKey{Namespace: namespace, Name: rp.Name}
-			if !apierrors.IsNotFound(err) {
-				kcLogger.Error(err, "Failed to delete configmap", "ConfigMap", key,
-					"Context", dcTemplate.K8sContext)
-				hasErrors = true
-			}
-		}
-	}
-	return
-}
-
-func (r *K8ssandraClusterReconciler) deleteServices(
+// setDcOwnership loads the remote resource identified by controlledKey, sets dc as its owner, and writes it back. If
+// the remote resource does not exist, this is a no-op.
+func setDcOwnership[T client.Object](
```

Comment:
It is a no-op:
```go
	if err := remoteClient.Get(ctx, controlledKey, controlled); err != nil {
		if errors.IsNotFound(err) {
			return result.Continue()
		}
```
I did it that way because it's not always trivial to determine if the resource is supposed to exist (based on the K8ssandraClusterSpec); we'd have to duplicate logic from the creation code.

All those resources are required for the CassandraDatacenter to start correctly, so I think we can safely assume that they exist.

Reply:
I have reservations about this because if the resource name changes (e.g. due to cluster name overrides or some other weird piece of logic) we could easily end up in a situation where the resources aren't marked with an owner correctly, and this failure would happen silently.